### ISSUE TITLE
python-requests obsolete on Fedora

### ISSUE THREAD

**Issue by mauved** [saltstack/salt-bootstrap#875]
### Description of Issue/Question

Running the bootstrap script fails on a fresh Fedora 23 installation since it looks like the package `python-requests` was split up into `python2-requests` and `python3-requests`.

```
Last metadata expiration check: 0:12:21 ago on Thu Jun 16 22:25:35 2016.
Package PyYAML-3.11-11.fc23.x86_64 is already installed, skipping.
Package libyaml-0.1.6-7.fc23.x86_64 is already installed, skipping.
Package python-crypto-2.6.1-7.fc23.x86_64 is already installed, skipping.
Error: installed package python2-requests-2.10.0-2.fc23.noarch obsoletes python-requests < 2.10.0-2.fc23 provided by python-requests-2.7.0-7.fc23.noarch
(try to add '--allowerasing' to command line to replace conflicting packages)
 * ERROR: Failed to run install_fedora_deps()!!!
```

I was able to get Salt bootstrapped on this system my modifying line 3176 and replacing `python-requests` with `python2-requests`.
### Versions and Systems

```
[root@salty ~]# ./bootstrap-salt.sh -v
./bootstrap-salt.sh -- Version 2016.05.11
[root@salty ~]# cat /etc/fedora-release 
Fedora release 23 (Twenty Three)
```

**Comment by jfindlay** [comment]
@mauved, thanks for reporting.

**Comment by rallytime** [comment]
@mauved Can you tell me how you're running the bootstrap script? (Any other options you're passing in?) I just ran the bootstrap script a couple of times on two Fedora 23 boxes and I am having trouble reproducing this error. I ran it first without any options, and then I ran it to install via git (`sh install_salt.sh -P git v2016.3.1`).

Here's my information from the last run:

```
[root@fedora ~]# sh install_salt.sh -v
install_salt.sh -- Version 2016.06.27
[root@fedora ~]# cat /etc/fedora-release
Fedora release 23 (Twenty Three)
[root@fedora ~]# salt-call --local test.versions_report
local:
    Salt Version:
               Salt: 2016.3.1

    Dependency Versions:
               cffi: 1.4.2
           cherrypy: Not Installed
           dateutil: 2.4.2
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.8
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: 0.21.1
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.6
       mysql-python: Not Installed
          pycparser: 2.14
           pycrypto: 2.6.1
             pygit2: Not Installed
             Python: 2.7.11 (default, Mar 31 2016, 20:46:51)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 14.7.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.3
                ZMQ: 4.1.2

    System Versions:
               dist: fedora 23 Twenty Three
            machine: x86_64
            release: 4.5.5-x86_64-linode69
             system: Linux
            version: Fedora 23 Twenty Three
```

**Comment by mauved** [comment]
Hi @rallytime! Here's a full run.

```
[root@salty ~]# cat /etc/fedora-release
Fedora release 23 (Twenty Three)
[root@salty ~]# ./salt-bootstrap.sh -v
./salt-bootstrap.sh -- Version 2016.06.27
[root@salty ~]# ./salt-bootstrap.sh
 *  INFO: Running version: 2016.06.27
 *  INFO: Executed by: -
 *  INFO: Command line: "./salt-bootstrap.sh "

 *  INFO: System Information:
 *  INFO:   CPU:          AuthenticAMD
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   4.5.6-200.fc23.x86_64
 *  INFO:   Distribution: Fedora 23

 *  INFO: Installing minion
 *  INFO: Found function install_fedora_deps
 *  INFO: Found function install_fedora_stable
 *  INFO: Found function install_fedora_stable_post
 *  INFO: Found function install_fedora_restart_daemons
 *  INFO: Found function daemons_running
 *  INFO: Found function install_fedora_check_services
 *  INFO: Running install_fedora_deps()
 *  INFO: Adding SaltStack's COPR repository
Last metadata expiration check: 0:10:10 ago on Tue Jul  5 23:04:31 2016.
Package PyYAML-3.11-11.fc23.x86_64 is already installed, skipping.
Package libyaml-0.1.6-7.fc23.x86_64 is already installed, skipping.
Package python-crypto-2.6.1-7.fc23.x86_64 is already installed, skipping.
Error: installed package python2-requests-2.10.0-2.fc23.noarch obsoletes python-requests < 2.10.0-2.fc23 provided by python-requests-2.7.0-7.fc23.noarch
(try to add '--allowerasing' to command line to
```

**Comment by rallytime** [comment]
Huh. Weird. I wonder why I am not hitting that same error. Can you run that again with the `-D` option, just in case there's anything useful in the debug logs?

**Comment by rallytime** [comment]
@mauved ^

**Comment by mauved** [comment]
Here's a run with `-D`.

```
[root@salty ~]# ./bootstrap-salt.sh -D
 *  INFO: Running version: 2016.06.27
 *  INFO: Executed by: /bin/sh
 *  INFO: Command line: "./bootstrap-salt.sh -D"

 *  INFO: System Information:
 *  INFO:   CPU:          AuthenticAMD
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   4.5.6-200.fc23.x86_64
 *  INFO:   Distribution: Fedora 23

 * DEBUG: Binaries will be searched using the following $PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin
 *  INFO: Installing minion
 * DEBUG: install_fedora_23_stable_deps not found....
 * DEBUG: install_fedora_23_deps not found....
 * DEBUG: install_fedora_stable_deps not found....
 *  INFO: Found function install_fedora_deps
 * DEBUG: DEPS_INSTALL_FUNC=install_fedora_deps
 * DEBUG: CONFIG_SALT_FUNC=null
 * DEBUG: PRESEED_MASTER_FUNC=null
 * DEBUG: install_fedora_23_stable not found....
 *  INFO: Found function install_fedora_stable
 * DEBUG: INSTALL_FUNC=install_fedora_stable
 * DEBUG: install_fedora_23_stable_post not found....
 * DEBUG: install_fedora_23_post not found....
 *  INFO: Found function install_fedora_stable_post
 * DEBUG: POST_INSTALL_FUNC=install_fedora_stable_post
 * DEBUG: install_fedora_23_stable_restart_daemons not found....
 * DEBUG: install_fedora_23_restart_daemons not found....
 * DEBUG: install_fedora_stable_restart_daemons not found....
 *  INFO: Found function install_fedora_restart_daemons
 * DEBUG: STARTDAEMONS_INSTALL_FUNC=install_fedora_restart_daemons
 * DEBUG: daemons_running_fedora_23_stable not found....
 * DEBUG: daemons_running_fedora_23 not found....
 * DEBUG: daemons_running_fedora_stable not found....
 * DEBUG: daemons_running_fedora not found....
 *  INFO: Found function daemons_running
 * DEBUG: DAEMONS_RUNNING_FUNC=daemons_running
 * DEBUG: install_fedora_23_stable_check_services not found....
 * DEBUG: install_fedora_23_check_services not found....
 * DEBUG: install_fedora_stable_check_services not found....
 *  INFO: Found function install_fedora_check_services
 * DEBUG: CHECK_SERVICES_FUNC=install_fedora_check_services
 *  INFO: Running install_fedora_deps()
 *  INFO: Adding SaltStack's COPR repository
Last metadata expiration check: 0:06:53 ago on Tue Jul  5 23:26:32 2016.
Package PyYAML-3.11-11.fc23.x86_64 is already installed, skipping.
Package libyaml-0.1.6-7.fc23.x86_64 is already installed, skipping.
Package python-crypto-2.6.1-7.fc23.x86_64 is already installed, skipping.
Error: installed package python2-requests-2.10.0-2.fc23.noarch obsoletes python-requests < 2.10.0-2.fc23 provided by python-requests-2.7.0-7.fc23.noarch
(try to add '--allowerasing' to command line to replace conflicting packages)
 * ERROR: Failed to run install_fedora_deps()!!!
 * DEBUG: Removing the logging pipe /tmp/bootstrap-salt.logpipe
```

**Comment by rallytime** [comment]
Thank you! I'll take another look at this tomorrow. The fix should be simple, but I wanted to reproduce it first.

**Comment by rallytime** [comment]
@mauved The fix in #909 should resolve this issue for you. Give that a try and let me know how it goes.
